extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts

```diff
@@ -743,6 +744,7 @@ export class PublishDatabaseDialog {
 
 	private createSqlCmdTable(view: azdataType.ModelView): azdataType.DeclarativeTableComponent {
 		this.sqlCmdVars = { ...this.project.sqlCmdVariables };
+		this.originalSqlCmdVarValues = { ... this.project.sqlCmdVariables };
```

Comment:
could also compare to `this.project.sqlCmdVariables` instead of having `this.originalSqlCmdVarValues`, unless there's a benefit to keep track of them separately

Reply:
IIRC, we would need to keep track of it separately, because we might have values from the publish profile as well.

Reply:
I was looking at the below added logic to figure out the same. May be worth figuring out how the three source of values (sqlcmd vars values in project, in profiles and manually updated) work together here.

Reply:
I've experimented a bit more with SSDT and added info about how SSDT works to #15990 in [this comment](https://github.com/microsoft/azuredatastudio/issues/15990#issuecomment-1492710062).

Reply:
Okay, the behavior should now match SSDT, plus the button disabling behavior.  Gif in the PR description is updated.